### ISSUE TITLE
WRITE-ERROR: EVENT=ITF,CC=41,SENSE=0040 Hercules 4.3.0.10296

### ISSUE THREAD

**Issue by abatlat** [SDL-Hercules-390/hyperion#387]
Hello All,

On Hercules **4.3.0.10296-SDL-ga4db8213** with Windows-6.2.9200 Professional 64-bit, when I run one utility I get one permanent I/O error with the error below:
```
 WRITE-ERROR: EVENT=ITF,CC=41,SENSE=0040,CSW=00079468  0E005FFE,@CP=079450,SEEK=00000001 AD000002
 WRITE-ERROR: @IOB=00079978,@DCB=000799C0,@DEB=8C50EC  EXTENT NUM=000,STR-CC.HH=01AD.0000,END-C.HH=0594.000E,NUMBER-OF-TRACKS=15,000
```
I've changed the disk type defining on 3390 model 1 without any compression, but the situation didn't change. So I've ipled my z/OS using Hercules 4.0.0 RC0 using the same utility and no error is raised on the same dasds.

Below follow portion of my config file:

```
 FACILITY ENABLE 073_TRANSACT_EXEC
 FACILITY ENABLE 050_CONSTR_TRANSACT
 MAXCPU 4
 ENGINES CP,CP,CP,IP
 NUMCPU 4
 AUTO_SCSI_MOUNT NO
 CCKD  RA=9,RAQ=16,RAT=16,WR=8,GCINT=5,GCPARM=4,NOSTRESS=0,TRACE=0,FREEPEND=-1
 CNSLPORT 3270
 CONKPALV (3,1,10)
 #
 DEVTMAX 0
 DIAG8CMD ENABLE
 ECPSVM NO
CODEPAGE default
 CPUMODEL 2827
 CPUSERIAL BADF35
 HTTP PORT 80 NOAUTH
 MODPATH .
 LOADPARM 0A8200..
 LPARNUM 01
LPARNAME HERCULES
 MAINSIZE 4096
MANUFACTURER HRC
MODEL 501
MOUNTED_TAPE_REINIT DISALLOW
 AUTOMOUNT ADD TAPES
 OSTAILOR Z/OS
 PANOPT FULLPATH
 PANRATE 500
 PGMPRDOS LICENSED
 PLANT ZZ
 SHCMDOPT NODIAG8
 SYSEPOCH 1900
 TIMERINT 50
 TRACEOPT TRADITIONAL
 TZOFFSET +1400
HERCPRIO 0
 TODPRIO -20
 DEVPRIO 8
 CPUPRIO 15
```

 Someone can help me with that, please?

 Bye ANGELO

**Comment by Fish-Git** [comment]
More information is needed. &nbsp; Please refer to: "[**Submitting problem reports**](https://github.com/SDL-Hercules-390/hyperion/issues/373)".

* Your _COMPLETE_ configuration file. (The portion you provided is not enough.)
* Your _COMPLETE_ Hercules log. (It might contain some important clues.)
* What guest you are running? You mention z/OS, but you don't say which version. Are you running z/OS 1.10? Or perhaps z/OS 2.4? Etc.
* What z/OS "utility" are you trying to run? May we see the JCL you submitted? May we see the resulting printout?
* Your error message is rather vague. Where does it appear? On the z/OS Master console? Or in your job's printout? Are there any other messages issued before the error messages? May we see the _COMPLETE_ job log? (Or _COMPLETE_ system log?)
* Can you explain how we can reproduce the problem for ourself? If we can recreate the same problem on our own test system, then we would have a much greater chance of finding your problem.

**Bottom line:** You have not given us enough information to determine what the problem might be, ANGELO. You need to provide much more (and much more _detailed_) information. Please provide additional inormation.

**Comment by abatlat** [comment]
Thanks, Fish for prompt replay 

z/OS 1.11    (the situation is the same on z/os 2.3)  
the utility used  is  Format  DMS FILe   CADISK  product  the jcl below 
//* JCL TO INITIALIZE (FORMAT) THE CA DISK                *    
//* FILES DATA SET.                                       *    
//*************************************************************
//FILEI   EXEC PGM=ADSMI000,PARM=ADSDM100,REGION=2048K         
//STEPLIB   DD DISP=SHR,DSN=DMS.RNN.SPER.LOADLIB               
//ABNLDUMP  DD DUMMY                                           
//CMDPRINT  DD SYSOUT=*                                        
//FILES     DD DISP=(,CATLG,DELETE),                           
//             DSN=DMS.RNN.FILES.NEW1,                         
//             VOL=SER=PUBI54,            <=== SUPPLY VOLUME   
//             UNIT=SYSDA,                                     
//             DCB=(DSORG=DA),                                 
//             SPACE=(CYL,10,,CONTIG)                          
//MSGPRINT  DD SYSOUT=*                                        
//PARMLIB   DD DISP=SHR,DSN=DMS.RNN.SPER.PARMLIB               
//SYSPRINT  DD SYSOUT=*                                        
//SYSUDUMP  DD SYSOUT=*                

  
 **### JCL JOB LOG**   ATTACHED
[JCL-SYSOUT.txt](https://github.com/SDL-Hercules-390/hyperion/files/6251314/JCL-SYSOUT.txt)

HERCULES CONFIG FILE
[CONFIGZ11.txt](https://github.com/SDL-Hercules-390/hyperion/files/6251316/CONFIGZ11.txt)

Hercules  LOG FILE
[Hercules Logfile.011.txt](https://github.com/SDL-Hercules-390/hyperion/files/6251324/Hercules.Logfile.011.txt)

              

many thanks

**Comment by Fish-Git** [comment]
Angelo,

I think I see your problem.

All of your 3390-9, 3390-27 and 3390-54 volumes are only 32-bit CCKD compressed dasd images instead of 64-bit CCKD64 compressed dasd images, and one of them (SSMS00 = 0AA7) has already reached its imposed maximum physical file size limit for that particular Hercules dasd image file format:

<pre>
HHC00333I 0:0AA7   <i><b>32/64       size free</b></i>  nbr st   reads  writes l2reads    hits switches
HHC00334I 0:0AA7                                                      readaheads   misses
HHC00335I 0:0AA7 ------------------------------------------------------------------------
HHC00336I 0:0AA7 [*] <i><b>32 04294937035 013%</b></i> 0016    0000131 0008782 0004795 0000559  0000637
HHC00337I 0:0AA7                                                         0000054  0000032
HHC00338I 0:0AA7 C:/DISCOE/ZOS111/DASD/DISKI/<i><b>SSMS00</b></i>
HHC00339I 0:0AA7 [0] <i><b>32 04294937035 000%</b></i> 0016 rw 0000131 0008782 0004795
</pre>

The Hercules 32-bit "CCKD" compressed dasd image file format is limited to a _maximum_ file size of slightly less than 4GB. Volume `SSMS00` is bumping up against that limit.

Now it may well be true that device 0AA7 = SSMS00 is defined as a model 3390-9 which theoretically should be able to hold up to 8GB of data, but it was created as only a Hercules CCKD (i.e. 32-bit) format dasd image file, which as already stated, is designed to only hold a maximum of 4GB of data. Once that limit is reached, "Bad Things" start to happen, as you are now discovering.

Version 4.2 of SDL Hyperion introduced support for a new **[64-bit Compressed CKD dasd image file format called CCKD64](https://sdl-hercules-390.github.io/html/hercrnot.html#4.2)**.

To fix your problem (and to prevent any such similar problems in the future), you should to convert _**ALL**_ of your  existing Hercules dasds to CCKD64 format instead, so that they will then be able to hold _ALL_ of the data that each 3390 model was designed to hold. The procedure for doing so is documented in our [**"Hercules CCKD64 Support" README**](https://github.com/SDL-Hercules-390/hyperion/blob/master/readme/README.CCKD64.md#Procedure). Please be sure to follow it closely. Step <span>#</span>1 (running `cckdcdsk -3` on your existing dasds) is _especially_ important!

If you need any further help with converting your dasds over to the new CCKD64 format, please post your question to the Hercules-390 support forum:

* **https://hercules-390.groups.io/g/group**

The folks there should be able to help you.

Good luck!
